generate/generate.go

```diff
@@ -751,6 +923,149 @@ func removeStackGeneratedFiles(
 	return removedFiles, nil
 }
 
+func generateRootFiles(
+	cfg *config.Tree,
+	genfiles []GenFile,
+	report *Report,
+) {
+	logger := log.With().
+		Str("action", "generate.generateRootFiles()").
+		Logger()
+
+	diskFiles := map[string]string{}       // files already on disk
+	mustExistFiles := map[string]GenFile{} // files that must be present on disk
+	mustDeleteFiles := map[string]bool{}   // files to be deleted
+
+	// this computes the files that must be present on disk after generate
+	// returns. They should not be touched if they already exist on disk and
+	// have the same content.
+	for _, file := range genfiles {
+		if file.Condition() {
+			logger := genFileLogger(logger, file)
+			logger.Debug().Msg("file must be generated (if needed)")
+
+			mustExistFiles[file.Label()] = file
+		}
+	}
+
+	// this computes the files that must be deleted if they are present on disk
+	for _, file := range genfiles {
+		if _, ok := mustExistFiles[file.Label()]; !ok {
+			logger := genFileLogger(logger, file)
+			logger.Debug().Msg("file must be deleted (if needed)")
+
+			mustDeleteFiles[file.Label()] = true
+		}
+	}
+
+	// reads the content of must-exist files (if they are present).
+	for label := range mustExistFiles {
+		logger := logger.With().Str("file", label).Logger()
+
+		logger.Debug().Msg("reading the content of the file on disk")
+
+		abspath := filepath.Join(cfg.RootDir(), label)
+		dir := path.Dir(label)
+		body, err := os.ReadFile(abspath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				logger.Debug().Msg("file do not exists")
+
+				continue
+			}
+			dirReport := dirReport{}
+			dirReport.err = errors.E(err, "reading generated file")
+			report.addDirReport(project.NewPath(dir), dirReport)
+			return
+		}
+
+		logger.Debug().Msg("file content read successfully")
+
+		diskFiles[label] = string(body)
+	}
+
+	// this deletes the files that exist but have condition=false.
+	for label, mustDelete := range mustDeleteFiles {
+		logger := logger.With().Str("file", label).Logger()
+
+		if mustDelete {
```

Comment:
do we need a map to boolean + an if here ? It seems that mustDeleteFiles is just a set of files that must be deleted, why would any entry be on the map and not be supposed to be deleted ? So it could be just map[string]struct{}, and then here we could just iterate it and delete everything (unless we do have a few entries with bool = false, but I dont think that is the case).

Reply:
I thought it'd be clearer with the bool but yeah it makes sense to just use struct{}.
thanks